relay-event-normalization/src/normalize/nel.rs

```diff
@@ -40,22 +223,202 @@ pub fn create_log(nel: Annotated<NetworkReportRaw>, received_at: DateTime<Utc>)
         }};
     }
 
-    add_attribute!("todo.nel.referrer", body.referrer);
-    add_attribute!(
-        "todo.nel.server_ip",
-        body.server_ip.map_value(|s| s.to_string())
-    );
-    add_attribute!("todo.nel.elapsed_time", body.elapsed_time);
-    add_attribute!("todo.nel.error_type", body.ty);
-    add_attribute!("todo.nel.phase", body.phase.map_value(|s| s.to_string()));
-    add_attribute!("todo.nel.sampling_fraction", body.sampling_fraction);
+    macro_rules! add_string_attribute {
+        ($name:literal, $value:expr) => {{
+            let val = $value.to_string();
+            if !val.is_empty() {
+                attributes.insert($name.to_owned(), val);
+            }
+        }};
+    }
+
+    add_string_attribute!("sentry.origin", "auto.http.browser_report.nel");
+    add_string_attribute!("browser.report.type", "network-error");
+
+    // Handle URL and extract server address if available
+    add_attribute!("url.full", raw_report.url.clone());
+    if let Some(url_str) = raw_report.url.value() {
+        let url_domain = extract_server_address(url_str);
+        add_string_attribute!("url.domain", &url_domain);
+    }
+    add_attribute!("http.request.duration", body.elapsed_time);
+    add_attribute!("http.request.method", body.method);
+    add_attribute!("http.request.header.referer", body.referrer.clone());
+    add_attribute!("http.response.status_code", body.status_code);
+    // Split protocol into name and version components
+    if let Some(protocol) = body.protocol.value() {
+        let parts: Vec<&str> = protocol.split('/').collect();
+        if !parts.is_empty() && !parts[0].is_empty() {
+            // e.g. "http"
+            add_string_attribute!("network.protocol.name", parts[0]);
+            if parts.len() > 1 && !parts[1].is_empty() {
+                // e.g. "1.1"
+                add_string_attribute!("network.protocol.version", parts[1]);
+            }
+        }
```

Comment:
If the protocol should always contain a `/` then `str.split_once()` is also an option, but your code currently deals with the version being optional, so I kept that logic the same.

```suggestion
        let mut parts = protocol.split('/');
        if Some(protocol) = parts.next() && !protocol.is_empty() {
            // e.g. "http"
            add_string_attribute!("network.protocol.name", protocol);
        }
        if Some(version) = parts.next() && !version.is_empty() {
                // e.g. "1.1"
            add_string_attribute!("network.protocol.version", version);
        }
```